subscriptions.md

```diff
@@ -0,0 +1,297 @@
+      "_links" : {
+        "dSpaceObject" : {
+          "href" : "http://localhost/api/core/subscriptions/1/dSpaceObject"
+        },
+        "ePerson" : {
+          "href" : "http://localhost/api/core/subscriptions/1/ePerson"
+        },
+        "self" : {
+          "href" : "http://localhost/api/core/subscriptions/1"
+        }
+      }
+    } ]
+  },
+  "_links" : {
+    "self" : {
+      "href" : "http://localhost/api/core/subscriptions"
+    },
+    "search" : {
+      "href" : "http://localhost/api/core/subscriptions/search"
+    }
+  },
+  "page" : {
+    "size" : 20,
+    "totalElements" : 1,
+    "totalPages" : 1,
+    "number" : 0
+  }
+}
+```
+
+Attributes
+* page, size [see pagination](README.md#Pagination)
+
+Return codes:
+* 200 OK - if the operation succeed
+* 401 Unauthorized - if you are not authenticated. Please note that this also apply to resource policy related to the Anonymous group
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators, users with ADMIN right can access the subscriptions
+
+## Single Subscription Object
+**GET /api/core/subscription/<:id>**
+
+Provide detailed information about a specific subscription.
+The JSON response document is as follow
+
+```
+{
+  "id" : 1,
+  "type" : "subscription",
+  "subscriptionParameterList" : [ {
+    "id" : 1,
+    "name" : "Frequency",
+    "value" : "Daily"
+  } ],
+  "subscriptionType" : "TestType",
+  "_links" : {
+    "dSpaceObject" : {
+      "href" : "http://localhost/api/core/subscriptions/1/dSpaceObject"
+    },
+    "ePerson" : {
+      "href" : "http://localhost/api/core/subscriptions/1/ePerson"
+    },
+    "self" : {
+      "href" : "http://localhost/api/core/subscriptions/1"
+    }
+  }
+}
+```
+
+Status codes:
+* 200 OK - if the subscription is found and it is visible to the current user.
+* 401 Unauthorized - if you are not authenticated and the subscription is not visible to anonymous users
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only Admin and owner subscription has access.
+* 404 Not found - if the subscription doesn't exist
+
+### Search methods
+#### findByEPerson
+**GET /api/core/subscription/search/findByEPerson?uuid=<:ePerson-uuid>**
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* uuid: mandatory, the uuid of the eperson that benefit of content subscription
+
+```json
+{
+  "_embedded": {
+    "subscriptions": [
+      {
+        "id": 60,
+        "type": "subscription",
+        "subscriptionParameterList" : [ {
+			    "id" : 2,
+			    "name" : "Frequency",
+			    "value" : "Daily"
+		  } ],
+        "subscriptionType": "subscription",
+        "_links": {
+          "dSpaceObject": {
+            "href": "http://localhost:8080/server/api/core/subscriptions/60/dSpaceObject"
+          },
+          "ePerson": {
+            "href": "http://localhost:8080/server/api/core/subscriptions/60/ePerson"
```

Comment:
We should just call this "eperson" (no camel case) as that's what other endpoints use. 

This same comment applies to everywhere you are using "ePerson" links in the REST API

Reply:
I suppose that we have let `ePerson` leak out in other parts of the contract?  Otherwise we might want to abstract it as suggested for `dspaceObject`:  "subscriber" or some such.

Reply:
`eperson` is already used throughout the REST Contract.  For example, we have an `eperson` endpoint: https://github.com/DSpace/RestContract/blob/main/epersons.md

However, I again stress that we are always using `eperson` in all lowercase... never in camel case. 

Mostly, I'm just looking for consistency across our REST API.  It's really confusing to external developers if we use different terminology across different endpoints, and even different capitalization, etc.   New endpoints should always reference existing ones to ensure they align with what already exists.